### ISSUE TITLE
Feature request: show AUR comments before install

### ISSUE THREAD

**Issue by palb91** [Jguer/yay#492]
#### Affected Version
```console
$ yay -V
yay v6.786.r90.g1002a4f
```

#### Issue

I compare some AUR helpers since few days – since I don't know what is happening with `yaourt` – and I choose `yay` for some intersting features (reverse ordering when searching). One feature is missing that I appreciated in yaourt, the possibility to see **AUR comments**.

Some packages are broken, and sometimes a quick fix is provided in comments, that's really help…

I imagine a config parameter such as:
```json
{ "aurcomments": true }
```

What do you think about that ?

palb91

**Comment by Jguer** [comment]
Seems somewhat out of scope for `yay`, news is already on the borderline but if a quickfix is available in the AUR wouldn't the first thing to do be going there and check, it's not that much of a time saver

**Comment by Jguer** [comment]
Depends, are the comments parseable or by XML? Because I think I saw it once implemented by page parsing and that's something I would like to avoid

**Comment by Morganamilo** [comment]
Nope pretty sure you have to parse the comments of the AUR page. Not that I've really looked into it.

**Comment by Jguer** [comment]
Zawinski's law at work I guess.

I'll try and look into it but although go offers great facilities in web parsing, I would really, really like to avoid turning `yay` into a web browser.

**Comment by palb91** [comment]
Hi,
> That's why I initially suggested using -W/--web for news.

Yep, I spend few time in open/close issues searching for "comments" and similar word… I haven't found this one :/ Sorry for duplicate…
At least #384 provides a workaround for this.

> Zawinski's law at work I guess.

Maybe, not sure where are the limits… I asked for that because yay is « based on the design of yaourt, apacman and pacaur » according to you presentation, and AUR comments was one of the reasons I used yaourt and never changed.

I understand your reluctance for what could imply this functionnality but eh… it doesn't hurt to ask :)

**Comment by sandwm** [comment]
I don't know the AurJson RPC interface, so I can't tell if that would do, but if web parsing would be necessary, pakku seems to implement it [here in downloadAurComments](https://github.com/kitsunyan/pakku/blob/master/src/aur.nim#L147). It's not too complicated, though I understand the reluctance to crawl the web instead of a well defined RPC interface.

When installing new AUR packages I usually go to their web page anyway, but when updating I like to see the (new) comments about the current version, so this feature would be very much appreciated. In case you decide not to implement it, I would like to know if pull requests for this feature would be accepted (although I can't promise if / when I would find the time).

In any case, thanks for your work! :)

**Comment by palb91** [comment]
Hi !
I spend few time on AUR helpers, it seems that all of them parse the HTML in order to get comments.
AurJson RPC interface does not provide comments, but it was a good idea @sandwm …

So if you @Jguer choose to not implement it, lets close this issue, and I will continue to use ``yay`` because indeed, you did a great job !

**Comment by Jguer** [comment]
Thinking about it, if we manage not to pull any external library for parsing. The structure is predictable, `http` is already pulled in the project, a simple parser can be written. It will just be hacky in the way that layout changes will break comment parsing.

@sandwm if you want to implement it, I would be willing to accept it, I think @Morganamilo would not object either

**Comment by sidhanthholalkere** [comment]
@sandwm are you still working on this? I can try to implement it if otherwise
Also @Jguer would I be able to use `ioutil` as well?

**Comment by Morganamilo** [comment]
You may just want to look at aur-talk. I've heard good things about it.

**Comment by sandwm** [comment]
@sidhanthholalkere no, I'm not working on it, but I would very much appreciate it if you would find the time.

**Comment by Zerophase** [comment]
It might be possible to pull out yaourts web parsing code for displaying aur comments. Might even be easier to implement now, as the aur now supports markup syntax. You only need the first x amount of comments to tell if there is a bigger issue to intervene manually, before breaking your system.

**Comment by fam4r** [comment]
I subscribe with that request.

Moreover IMHO the use-case of opening the (every) related AUR web page while updating a package is very uncomfortable.

I also want to contribute quoting @Morganamilo reference to `aur-talk` linking to the source code which can help who wants to implement that avoiding you the effort of searching stuff

This https://github.com/GermainZ/aur-talk/blob/master/aur_talk.py#L34 could be a  good start

**Comment by drws** [comment]
If this idea is implemented one day I'd like to propose an additional filter to only display recent AUR comments. Filtering based solely on comment age doesn't cover all cases. I was thinking about displaying all comments since the last update AND/OR a fixed amount of time, possibly configurable in days.

**Comment by andruska** [comment]
Searching for command line AUR comments led me here. I also support the feature displaying comments. This would save me from having to launch a browser for it and in turn start looking for comments there.

**Comment by aksh1618** [comment]
Just chipping in as a years-long trizen user: being able to see AUR comments is my favourite trizen feature, and it has proved helpful more times than I can count: very often there are troubleshooting comments that folks leave when facing an issue while installing something, and it has been very helpful to be able to scroll up and see the issue I'm facing being already addressed in the comments. And this is on top of the fact that it is not uncommon for AUR package maintainers to leave pinned comments with FAQ or some other context.

Being able to see all these things right in the terminal while installing is very convenient, and IMO a much more seamless experience than having to leave the terminal and going to the browser. As trizen is beginning to show age now, I'm considering switching to yay but this feature is quite hard to leave behind.

Maybe trizen's implementation of this feature can serve as a reference point?
